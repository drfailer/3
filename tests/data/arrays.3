nil test(chr str[100]) bgn
  shw(str)
  shw("\n")
  shw(str[2])
  shw("\n")
end

nil main() bgn
  int a[5]
  chr string[10]
  int b
  chr c

  ~~~ test assignment to string
  set(string, "hello")

  shw(string)
  shw("\n")
  test(string)
  test("hello")

  for b in rng(0, 5, 1) bgn
    set(a[b], b)
  end
  set(a[4], 97)

  ~~~ test assignment to c
  set(c, a[4])
  shw(c)
  shw("\n")
  for b in rng(0, 5, 1) bgn
    shw(a[b])
    shw("\n")
  end
end
