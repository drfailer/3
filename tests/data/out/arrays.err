[[1;33mWARN[0m]: ./data/arrays.3:28: in assignment, [1;34mc[0m is of type [1;34mchr[0m but the value assigned is of type [1;34mint[0m.

