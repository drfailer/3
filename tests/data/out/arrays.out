hello
hello
l
hello
l
a
0
1
2
3
97
